src/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/CompositionContainerTests.cs

```diff
@@ -3022,6 +3023,33 @@ public void TestExportedValueCachesNullValue()
             Assert.Null(container.GetExportedValue<string>("Property"));
         }
 
+        [Fact]
+        public void TestExportedValueUsingWhereClause_ExportSuccessful()
+        {
+            CompositionContainer container = new CompositionContainer(new AssemblyCatalog(typeof(AssemblyCatalogTestsReflectionContext).Assembly));
+            IMefCollection<DerivedClass, BaseClass> i = container.GetExportedValue<IMefCollection<DerivedClass, BaseClass>>("UsingWhereClause");
+            Assert.NotNull(i);
+        }
+
+        public interface IMefCollection<TC, TP> : IList<TC>, IMefCollection where TC : TP { }
+        public class BaseClass { }
+        public class DerivedClass : BaseClass { }
+
+        public interface IMefCollection
```

Comment:
for readability you might want to move this interface above the IMefCollection<TC, TP> interface, as it is the base interface.

Reply:
No you're right. I should remove. Thanks